modules/3d/src/opencl/tsdf_functions.cl

```diff
@@ -0,0 +1,20 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+
+// Partially rewritten from https://github.com/Nerei/kinfu_remake
+// Copyright(c) 2012, Anatoly Baksheev. All rights reserved.
+
+__kernel void preCalculationPixNorm (__global char * pixNormsPtr,
```

Comment:
This kernel probably slowdowns OpenCL computations:
- from N+M used memory is raised to N*M which doesn't fit into cache.
- latency of sqrt+2fma is smaller that DDR memory access

Reply:
Agree, this makes more sense to calculate this on CPU
Fixed
(it did not impose serious performance penalties: it's calculated only once, at 1st frame)